pkg/northbound/device/device.proto

```diff
@@ -41,15 +42,15 @@ message UpdateRequest {
 
 // UpdateResponse is sent in response to an UpdateDeviceRequest
 message UpdateResponse {
-    // metadata is the updated device metadata
-    ObjectMetadata metadata = 1;
+    // version is the updated version of the device
+    uint64 version = 1 [(gogoproto.casttype) = "Version"];
```

Comment:
is this semantic versioning e.g. 1.0.0 serialized in to one number?
Super magic powers going on here - I wonder how non go code (e.g. Typescript) would deserialize it?

Reply:
This is a revision version assigned by the store. It's used to prevent two concurrent changes from being made to the same version of the same object. This is the pattern the Kubernetes API uses to provide concurrency control as well. Basically, when an object is updated, the object is returned with a version number. Then, if the object is modified again, the modified object's version number is compared to the store to determine whether an earlier change to the same version was already made, and rejects the change if so, forcing the client to read the updated object state and retry. We can also rename this field `Revision` to avoid confusion.

The custom `Version` type is only relevant to Golang code generation. The `casttype` annotation just casts the `uint64` field to `Version` when it's read from the wire, and casts `Version` back to `uint64` when it's written. In other languages, the `gogoproto.casttype` annotation is unknown, so they will generate code that just operates on a `uint64` type.

Reply:
ok - thanks for the explaination